client/src/App.js

```diff
@@ -33,34 +33,46 @@ class App extends Component {
                 isYogaAdmin: false,
                 setAccess: this.setAccess,
 
-                lang: 'hu',
+                lang: localStorage.getItem('lang'),
                 dictionary: dictionaryList[localStorage.getItem('lang')],
                 changeLang: this.changeLang
             },
 
             dataContext: {
-                userName: 'unknown',
+                userName: {
+                    firstName: '',
+                    lastName: '',
+                    fullName: '',
+                    nameOrder: localStorage.getItem('lang') === 'hu' ? -1 : 1
+                },
                 avatarSrc: '/images/noAvatar.svg',
                 setUsername: this.setUsername,
                 setAvatarSrc: this.setAvatarSrc
             }
         };
     }
 
+    getFullName = (firstName, lastName) => {
+        const { lang } = this.state.uiContext;
+        return lang === 'hu' ? `${lastName} ${firstName}` : `${firstName} ${lastName}`;
+    }
+
+    getNameOrder = () => this.state.uiContext.lang === 'hu' ? -1 : 1;
+
     toggleHamburger = () => {
-        const uiContext = { ...this.state.uiContext };
+        const { uiContext } = this.state;
         uiContext.isHamburgerOpen = !this.state.uiContext.isHamburgerOpen;
```

Comment:
I think 'this.state." is redundant here in line 64.

Reply:
You are right, good catch!